clip_retrieval/clip_filter.py

```diff
@@ -16,9 +16,9 @@ def clip_filter(query, output_folder, indice_folder, num_results=100, threshold=
     import clip  # pylint: disable=import-outside-toplevel
 
     device = "cuda" if torch.cuda.is_available() else "cpu"
-    model, _ = clip.load("ViT-B/32", device=device, jit=False)
+    model, _ = clip.load(clip_model, device=device, jit=False)
 
-    data_dir = Path(indice_folder + "/metadata")
+    data_dir = Path(indice_folder + "/metadata") if metadata_folder is None else Path(metadata_folder)
```

Comment:
Since "index" option lets one omit the metadata copying, metadata may not be part of the indices. This change is to address that scenario.

I also wanted to change the clip_back.py as well similarly, but I couldn't come up with any elegant solution to include metadata configuration since we take the indices path from the json file where there could be multiple indices.

Reply:
metadata should always be present, it doesn't make sense not to have metadata: can't fetch the corresponding data from the ids returned by the index